modules/utility/preview/markdown-preview/markdown-preview.nix

```diff
@@ -0,0 +1,57 @@
+{lib, ...}: let
+  inherit (lib) types mkEnableOption mkOption;
+in {
+  options.vim.utility.preview = {
+    markdownPreview = {
+      enable = mkEnableOption "enable Markdown preview in neovim with markdown-preview.nvim";
+
+      autoStart = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Automatically open the preview window after entering a Markdown buffer";
+      };
+
+      autoClose = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Automatically close the preview window after leaving a Markdown buffer";
+      };
+
+      lazyRefresh = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Only update preview when saving or leaving insert mode";
+      };
+
+      filetypes = mkOption {
+        type = with types; listOf str;
+        default = ["markdown"];
+        description = "Allowed filetypes";
+      };
+
+      alwaysAllowPreview = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Allow preview on all filetypes";
+      };
+
+      broadcastServer = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Allow for outside and network wide connections";
+      };
+
+      customIP = mkOption {
+        type = types.str;
+        default = "";
+        description = "IP-address to use";
+      };
+
+      customPort = mkOption {
+        type = types.str;
+        default = "";
+        description = "Port to use";
```

Comment:
defaults being `""` seem to me like something that could cause problems if the user forgets to specify. Lets either provide working defaults (maybe upstream has one?) or append `let g:mkdp_open_ip = '${cfg.customIP}'` and `let g:mkdp_port = '${cfg.customPort}'` inside `optionalString evaluations

Reply:
This can be marked resolved if upstream plays nicely with empty string as defaults.

Reply:
Upstream defaults are just empty. If the value for port is empty, it will choose a random one. For the IP, it'll use ``localhost``unless something else is specified.